File: 242.有效的字母异位词.py
#
# @lc app=leetcode.cn id=242 lang=python
#
# [242] 有效的字母异位词


# @lc code=start
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != lent):
            return False
        for x in set(s):
            if s.count(x) !=t.count(x):
                return False
        return True
# @lc code=end

File: 242.有效的字母异位词.cpp
/*
 * @lc app=leetcode.cn id=242 lang=cpp
 *
 * [242] 有效的字母异位词
 */

// @lc code=start
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size())
            return false;
        // sort(s.begin(), s.end());
        // sort(t.begin(), t.end());
        // return s== t;
        unordered_map<char, int> res;
        for(char ch : s)
            res[ch] += 1;
        for(char ch : t)
        {
            if(res[ch])
                res[ch]--;
            else{
                return false;
            }

        }
        return true;
        

    }
};
// @lc code=end

